c_emulator/riscv_sim.c

```diff
@@ -287,17 +287,17 @@ static int process_args(int argc, char **argv)
       rv_enable_misaligned = true;
       break;
     case OPT_PMP_COUNT:
-      pmp_count = atol(optarg);
-      fprintf(stderr, "PMP count: %lld\n", pmp_count);
+      pmp_count = (uint8_t)atoi(optarg);
+      fprintf(stderr, "PMP count: %u\n", pmp_count);
       if (pmp_count != 0 && pmp_count != 16 && pmp_count != 64) {
         fprintf(stderr, "invalid PMP count: must be 0, 16 or 64");
         exit(1);
       }
       rv_pmp_count = pmp_count;
       break;
     case OPT_PMP_GRAIN:
-      pmp_grain = atol(optarg);
-      fprintf(stderr, "PMP grain: %lld\n", pmp_grain);
+      pmp_grain = (uint8_t)atoi(optarg);
```

Comment:
This seems wrong? You'll silently truncate larger values which is potentially very confusing.

Reply:
@Timmmm - would you like to fix the two warnings since you are most familiar with this code? I can close this PR if you provide a fix.

Reply:
Please do.

Reply:
> Touché :D
> 
> I would use @arichardson 's suggestion of `PRIu64`. It is very ugly so you could use `%llu` which is less ugly. I think people only recommend `PRIu64` due to theoretical fears about portability (maybe if `unsigned long long` was 128 bits??).
> 
> I can make a new PR if you like?

The problem is that some plaforms use `unsigned long` for uint64_t and some use `unsigned long long` (I believe macOS and glibc systems differ), so whichever one you use you'll get a warning on one platform.

Another option is to add a `(uintmax_t)` cast and the `%j` modifier, but that's just as many characters as the `PRIu64` macro.

Reply:
Ahh TIL! The things C programmers have to put up with... :-D

I'll make a PR with that.